addons/hr_holidays/tests/test_leave_requests.py

```diff
@@ -1146,3 +1117,19 @@ def test_duration_display_global_leave(self):
         msg = "hr_holidays: duration_display should update after adding an overlapping holiday"
         self.assertEqual(sick_leave.duration_display, '2 days', msg)
         self.assertEqual(comp_leave.duration_display, '8 hours', msg)
+
+    def test_holiday_type_allocation_requirement_edit(self):
+        # Does not raise an error since no leave of this type exists yet
+        self.holidays_type_1.requires_allocation = 'yes'
+
+        self.env['hr.leave'].create({
+            'name': 'Test leave',
+            'employee_id': self.employee_emp_id,
+            'holiday_status_id': self.holidays_type_1.id,
+            'date_from': (datetime.today() - relativedelta(days=1)),
+            'date_to': datetime.today(),
+            'number_of_days': 1,
+        })
```

Comment:
this shouldn't work as there is no allocation for that leave type :thinking:

Reply:
It looks to me like there is no actual check for allocations here which is interesting. You are functionally prevented from selecting a leave type for which you have no allocation so you couldn't even make that request from the client side in the first place.

Still, this is pretty weird and I agree that I shouldn't have done that in my test. I'll change it

Reply:
I don't know how it was working exactly in 16.0 though I thought there was some check :thinking: , but either way you would have gotten an error raised in the forwardports since I'm certain there is a check at that point

Reply:
:+1: